certs mgmt guide (#3307)
hyperledger/fabric: docs/source/certs_management.md

> ## Certificate Management Guide

Reviewer: Did you mean to use ## instead of # for the heading?
Having two ## in this document causes two topics to be added to the table of contents... not sure if that is what you intended?

Author: ah I was wondering about that toc effect, ty.

**Audience**: Hyperledger Fabric network admins

> This guide provides overview information and details for a network admin to manage certificates (certs) in Hyperledger Fabric.

Reviewer: ```suggestion
This guide provides overview information and details for a network administrator to manage certificates (certs) in Hyperledger Fabric.
```

Author: ok

### Prerequisites and Resources

> The following Fabric documentation resources on identities, Managed Service Providers (MSPs) and Certificate Authorities (CAs) provide context for understanding certificate management:

Reviewer: ```suggestion
The following Fabric documentation resources on identities, Membership Service Providers (MSPs) and Certificate Authorities (CAs) provide context for understanding certificate management:
```

> * [Identity in Hyperledger Fabric](https://hyperledger-fabric.readthedocs.io/en/latest/identity/identity.html#identity)

Reviewer: Use relative links instead of absolute links. This will preserve the release branch. e.g. latest should navigate to latest. release-2.5 should eventually navigate to release-2.5.

See a good example at the top of https://raw.githubusercontent.com/hyperledger/fabric/main/docs/source/install.md.

Author: ok

* [MSPs in Hyperledger Fabric](https://hyperledger-fabric.readthedocs.io/en/latest/membership/membership.html#membership-service-provider-msp)
* [Registration and Enrollment](https://hyperledger-fabric-ca.readthedocs.io/en/latest/deployguide/use_CA.html#overview-of-registration-and-enrollment)
* [Registering an Identity](https://hyperledger-fabric-ca.readthedocs.io/en/latest/deployguide/use_CA.html#register-an-identity)

> * [Enrolling an Identity](https://hyperledger-fabric-ca.readthedocs.io/en/latest/deployguide/use_CA.html#enroll-an-identity)

Reviewer: Add https://hyperledger-fabric.readthedocs.io/en/latest/msp.html to the list.

Author: ok

### Key Concepts

**Registration** – A username (enrollment ID) and password pair, stored in the Certificate Authority (CA). This registration is created by a CA admin user, has no expiration, and contains any required roles and attributes.

> **Enrollment** – An identity based on the registration username and password. The enrollment contains a public/private key pair and an X.509 certificate issued by the organization Certificate Authority (CA). The certificate encodes roles and attributes from the registration.

Reviewer: I wouldn't say an enrollment "is an identity". I would say enrollment is the process of generating a public/private key pair and an X.509 certificate that represents an identity (or something like that).

Author: Well, it's currently parallel with the "Registration" usage (describing the result not the process), so we either leave it as is or change both of them -- to describe a process and not the result. 
@jkneubuh ?

Author: Both can be read as a process, or as a result. Changing one to read as a process but not the other won't work.

Reviewer 2: The process / verb forms are **Register** and **Enroll**.   The noun forms, referring to the result of the activities are **A Registration** and **An Enrollment**.

Maybe - change the ordering, or lessen the emphasis on _identity_ in the latter?

**Enrollment** – An enrollment contains a public/private key pair and an X.509 certificate issued by the organization's Certificate Authority (CA). The certificate encodes roles, attributes, and metadata representing an identity within a Fabric network.  An enrollment is associated with a CA registration by username and password. 

?

Author: @jkneubuh your latest seems good. Describing both as an object / result not as a process.

**Identity** - A public certificate and its private key used for encryption. The public certificate is the X.509 certificate issued by the CA, while the private key is stored out of band, on a secure storage.

### Certificate Types

Hyperledger Fabric implements two types of certificates: 1) **Enrollment** certificates for identities and 2) **TLS** certificates for node and client communications.

#### Enrollment Certificates

Enrollment certificates are classed into four types:

* **Admin**
* **Peer**
* **Orderer**
* **Client**

Each enrollment certificate type has a specific role:

**Admin:** X.509 certificates used to authenticate admin identities, which are required to make changes to  Fabric configurations.

**Peer:** X.509 certificates used to enroll peer nodes, located physically on the node or mapped to the node. For a Fabric peer node to start, it must have a valid enrollment certificate with the required attributes.

**Orderer:** X.509 certificates used to enroll orderer nodes, located physically on the node or mapped to the node. For a Fabric orderer node to start, it must have a valid enrollment certificate with the required attributes.

> **Client:** X.509 certificates that allow signed requests to be passed from clients to Fabric nodes. These requests require a signature for admin level certificates.

Reviewer: I'm not sure what "These requests require a signature for admin level certificates" means here. What are we trying to convey?

Author: idk, @jkneubuh ?

Reviewer 2: Client certificates define the identity of a **client application** submitting transactions to the blockchain.

Author: " Client certs define the identities of client applications submitting transactions to a Fabric network."

#### TLS Certificates

TLS certificates allow Fabric nodes and clients to sign and encrypt communications. A valid TLS Certificate is required for any channel communication.

### Enrollment Certificate Expiration

> All Enrollment certificates are assigned an expiration date by the issuing Certificate Authority (CA).  Expiration dates must be monitored, and certificates must be re-enrolled before expiration. The most important Enrollment Certificate parameter is the **Not After** element, which indicates its expiration date.

Reviewer: Expiration applies to both Enrollment and TLS certificates.

Author: ok.

## Certificates and Locations

Organization CAs supply X.509 Enrollment Certificates for identities and the TLS CAs supply TLS Certificates for securing node and client communications.

### Organization CA Certificates

> Organization CA Root certificates and Organization CA Admin certificates are issued for each organization.

Reviewer: I wouldn't say root certificates are 'issued' in the traditional Fabric CA enrollment sense. They will be self-signed certificates if creating a new CA, or will be provided by an external certificate authority.

The admin certificate however is 'issued' in the traditional Fabric CA enrollment process.

Author: ok

#### Organization CA Root Certificate

**Description**: Public certificate that permits verification of all certificates issued by the Organization CA.

> **Location**: Stored on disk in the Organization CA directory, or in the channel configuration (ca-cert.pem).

Reviewer: ```suggestion
**Location**: Stored on disk in the Organization CA directory (ca-cert.pem), and also copied into the channel configuration to verify identifies for this organization.
```

Author: ok.

**Impact if expired**: A new Organization CA Root certificate must be issued. Organization CA Root certificates are valid for 15 years.

#### Organization CA Admin Certificate

**Description**: Certificate used when making admin requests to the Organization CA.

**Location**: Dependent on implementation:

<pre>
msp
 ├── IssuerPublicKey
 ├── IssuerRevocationPublicKey
 ├── cacerts
 │   └── localhost-7053.pem
 ├── keystore
 │   └── key.pem
 ├── signcerts
 │   └── cert.pem
 └── user

> </pre>

Reviewer: Before providing the first msp directory, we need to explain that each identity has a local msp directory structure, which contains their certificate in the signcerts directory and their private key in the keystore directory.
We should also link to https://hyperledger-fabric.readthedocs.io/en/latest/membership/membership.html#msp-structure for more details about the msp directory.

Author: ok

**Impact if expired**: Cannot register or enroll new identities, but transaction traffic does not stop.

> [Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-orderer-org-s-ca-admin)

Reviewer: Need a description of what this link refers to beyond just 'Reference'.

Author: ok

### TLS CA Certificates

> TLS CA Root certificates and TLS CA Admin Certificates are issued for each organization.

Reviewer: All the comments above for Organization CA Certificates also applies to TLS CA Certificates.

Author: ok.

#### TLS CA Root Certificate

**Description**: Certificate used when making admin requests to the TLS CA.

**Location**: Stored on disk in the TLS CA directory or in the channel configuration (ca-cert.pem).

**Impact if expired**: A new TLS CA Root certificate must be issued. TLS CA Root certificates are valid for 15 years.

#### TLS CA Admin Certificate

**Description**: Certificate used for admin requests to the TLS CA.

**Location**: Dependent on implementation:

<pre>
msp
 ├── IssuerPublicKey
 ├── IssuerRevocationPublicKey
 ├── cacerts
 │   └── localhost-7053.pem
 ├── keystore
 │   └── key.pem
 ├── signcerts
 │   └── cert.pem
 └── user
</pre>

> **Impact if expired**: Cannot register or enroll new identities, but transaction traffic does not stop.

Reviewer: 'identities' in this context is confusing as people generally think of identities as coming from the Organization CA.
We need to explain that TLS certificates are registered and enrolled in the same way as Organization identities. That is, the result of a TLS enrollment is a TLS certificate.

Author: Sure, that doesn't really describe the impact if expired though. Such as -- 

"All communication between the node and its clients and channels stops." @jkneubuh ? 

I think this one was just missed by the review, unintended.

Author: And an add to Key Concepts, an entry for TLS cert -- 

"**TLS** - A public Transport Layer Security (TLS) certificate that authorizes client and node communications. On Fabric, registration and enrollment are the same for identity certificates and TLS certificates.

Reviewer 2: In the Org CA section, the impact of an expiration is that the Org Administrator will no longer be able to register new identities with the CA.

In the TLS CA section, the impact is that the Fabric Administrator will no longer be able to issue or renew TLS certificates for nodes in the network.

Author: @jkneubuh both seem good thanks.

    │    ├── admincerts
    │    │   └── cert.pem
    │    ├── cacerts
    │    │   └── localhost-7053.pem
    │    ├── keystore
    │    │   └── key.pem
    │    ├── signcerts
    │    │   └── cert.pem
    │    └── user
    |── tls
</pre>

**Impact if expired**: Production outage. Peers do not start without a valid Enrollment certificate.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-peer1)

#### Peer TLS Certificate

**Description**: Authenticates node component communication on the channel.

**Location**: Dependent on implementation:

<pre>
org1ca/
└── peer1
    ├── msp
    └── tls
        ├── cacerts
        ├── keystore
        │   └── key.pem
        ├── signcerts
        │   └── cert.pem
        ├── tlscacerts
        │   └── tls-localhost-7053.pem
        └── user
</pre>

**Impact if expired**: Production outage. No communication to the peer is possible.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-peer1)

### Orderer Certificates

Orderer Enrollment certificates and Orderer TLS certificates are issued for each organization.

#### Orderer Enrollment Certificate

> **Description**: The public key that the orderer uses to sign transactions.

Reviewer: ```suggestion
**Description**: The public key that the orderer uses to sign blocks.
```

Author: ok

**Location**: Dependent on implementation:

<pre>
 └── orderer1
     ├── msp
     │   ├── admincerts
     │   │   └── cert.pem
     │   ├── cacerts
     │   │   └── localhost-7053.pem
     │   ├── keystore
     │   │   └── key.pem
     │   ├── signcerts
     │   │   └── cert.pem
     │   |── user
     └── tls
</pre>

**Impact if expired**: Production outage. Orderers do not start without a valid Enrollment certificate.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-orderer)

#### Orderer TLS Certificate

> **Description**: TLS certificate for the ordering node.

Reviewer: ```suggestion
**Description**: TLS certificate for the ordering node communication.
```

Author: ok

**Location**: Dependent on implementation:

<pre>
ordererca/
└── orderer1
    ├── msp
    └── tls
        ├── cacerts
        ├── keystore
        |   └── key.pem
        ├── signcerts
        │   └── cert.pem
        ├── tlscacerts
        │   └── tls-localhost-7053.pem
        └── user
  </pre>

**Impact if expired**: Production outage. Ordering nodes are no longer allowed to participate in cluster.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-orderer)

### Admin Certificates

Ordering Service Organization Channel Admin certificates and Peer Service Organization Channel Admin certificates are issued for each organization.

#### Ordering Service Organization Channel Admin Certificate

> **Description**: Admin certificate used to administer the ordering service and submit or approve channel updates. The associated admin identity is enrolled with the Organization CA before the node and Organization MSP are created.

Reviewer: First sentence - need to indicate that the admin is associated with an organization (not the ordering service or channel as a whole).
Second sentence - I don't know what this is trying to say, you don't create an admin identity before the Organization MSP...

Author: First sentence: "Certificate for an organization administrator to manage ordering service and channel updates." 
Second - deleted, this would have come from a technical tester giving input.

**Location**: Dependent on implementation:

<pre>
ordererca/
└── ordereradmin
└── msp
    ├── admincerts
    │   └── cert.pem
    ├── cacerts
    │   └── localhost-7053.pem
    ├── keystore
    │   └── key.pem
    ├── signcerts
    │   └── cert.pem
    └── user
</pre>

**Impact if expired**: Transactions can continue to work successfully. Cannot modify channels from a client application or manage the orderer from the console.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-org0-s-admin)

#### Peer Service Organization Channel Admin Certificate

> **Description** - Admin certificate used to administer the peer and install smart contracts on the peer. Can also be configured to manage the application channel that the peer belongs to. The associated admin user is registered with the Organization CA before the peer and Organization MSP are created.

Reviewer: Same comment as above.

Author: first: "Certificate for an organization administrator to manage a peer, including channel and chaincode services."
2nd and 3rd - deleted.

**Location** - Dependent on implementation:

<pre>
org1ca/
└── org1admin
└── msp
├── admincerts
│     └── cert.pem
├── cacerts
│     └── localhost-7053.pem
├── keystore
│     └── key.pem
├── signcerts
│     └── cert.pem
└── user
</pre>

**Impact if expired**: Transactions can continue to work successfully. Cannot install new smart contracts from a client application or manage the peer from the console.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-org1-s-admin)

### Client Certificates

**Description**: Three types of Client certificates are issued for each organization:

1. Certificates used to authenticate an identity to the Gateway when submitting transactions
2. Client-side TLS certificates used to establish a secure network link to a Gateway peer

> 3. Certificate for a client SDK user if no gateway is accessed.

Reviewer: There are two types of certificates issued to a client (same story as peers and orderers and we can use consistent language and structure).
1 - Organization enrollment certificate - authenticates the client identity when communicating with peers and orderers (this covers your 1 and 3).
2 - TLS certificate - authenticates client communications, only required if mutual TLS is configured.

Author: ok.

> Both the client and the CA have to sign a client certificate for it to be valid.

Reviewer: I don't know what this is trying to say, but it doesn't look correct.

Author: 319 deleted

Client certificates expire after one year, using the Hyperledger Fabric CA default settings. Client certificates can be re-enrolled using either command line Hyperledger Fabric CA utilities or the Fabric CA client SDK.

**Impact if expired**: Client certificates must be re-enrolled before expiration or the client application will not be able to connect to the Fabric nodes.

[Reference](https://hyperledger.github.io/fabric-sdk-node/release-2.2/FabricCAClient.html#reenroll__anchor)

### Certificate Decoding

> X.509 certificates are distinct from registrations, which do not have an expiration. X.509 certificates are created based on the registration and contain information from the CA. This information includes metadata indicating the parent CA and encoded information describing the purpose of the certificate.

Reviewer: "which do not have an expiration" looks mis-placed. There is no context for expiration discussion here. And it is strange that we only mention registration here. We should also mention enrollment as that is the process that actually creates the certificate based on the registration.

We should mention that the certificate also includes an expiration in the "Not After" field.

Author: ok

**Orderer:** X.509 Certificates used to enroll orderer nodes, located physically on the node or mapped to the node. For a Fabric orderer node to start, it must have a valid Enrollment Certificate with the required attributes.

**Client:** X.509 Certificates that allow signed requests to be passed from clients to Fabric nodes. Client certs define the identities of client applications submitting transactions to a Fabric network.

### TLS Certificates

TLS Certificates allow Fabric nodes and clients to sign and encrypt communications. A valid TLS Certificate is required for any channel communication.

### Certificate Expiration

Enrollment and TLS Certificates are assigned an expiration date by the issuing Certificate Authority (CA).  Expiration dates must be monitored, and certificates must be re-enrolled before expiration. The most important certificate parameter is the **Not After** element, which indicates its expiration date.

## Certificates and Locations

Organization CAs supply X.509 Enrollment Certificates for identities and the TLS CAs supply TLS Certificates for securing node and client communications.

### Organization CA Certificates

> Organization CA Root Certificates and Organization CA Admin Certificates are self-signed certificates if creating a new Certificate Authority (CA), or provided by an external CA.

Reviewer: This sentence is only true for Organization CA Root Certificates, not Organization CA Admin Certificates. Therefore I would move this sentence to the Description of the next section.

Author: ok. added some content to replace it to avoid an empty level 3 heading.

#### Organization CA Root Certificate

**Description**: Public Certificate that permits verification of all certificates issued by the Organization CA.

**Location**: Stored on disk in the Organization CA directory (ca-cert.pem), and copied into the channel configuration to verify identifies for the organization.

**Impact if expired**: A new Organization CA Root Certificate must be issued. Organization CA Root Certificates are valid for 15 years.

#### Organization CA Admin Certificate

**Description**: Certificate used when making admin requests to the Organization CA.

> **Location**: Dependent on implementation:

Reviewer: Colon here is confusing given that there is a Note under it. It would help to put the Note in a box or blockquote.

**Note**: Each identity has a local **msp** directory structure which contains its certificate in the **signcerts** directory and its private key in the **keystore** directory. For details on the **msp** directory, refer to [MSP Structure](https://hyperledger-fabric.readthedocs.io/en/latest/membership/membership.html#msp-structure).

<pre>
msp
 ├── IssuerPublicKey
 ├── IssuerRevocationPublicKey
 ├── cacerts
 │   └── localhost-7053.pem
 ├── keystore
 │   └── key.pem
 ├── signcerts
 │   └── cert.pem
 └── user
</pre>

> **Impact if expired**: The Organization Administrator cannot register or enroll new identities with the CA, but transaction traffic does not stop.

Reviewer: ```suggestion
**Impact if expired**: The Organization Administrator cannot register new identities with the CA, but transaction traffic does not stop.
```

Reviewer: This comment is not addressed yet.

Author: This one is done. There are so many comments that apparently GitHub puts some of them in "hidden conversations" which need to be located and expanded.

[Reference - Enroll Orderer Org’s CA Admin](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-orderer-org-s-ca-admin)

### TLS CA Certificates

> TLS CA Root Certificates and TLS CA Admin Certificates are self-signed certificates if creating a new Certificate Authority (CA), or provided by an external CA.

Reviewer: This sentence is only true for CA Root Certificates, not CA Admin Certificates. Therefore I would move this sentence to the Description of the next section.

Reviewer: This comment is not addressed yet.

Author: Done. ... So like the other one, this is the identical comment from the prior section, copy pasted. It's helpful to differentiate comments, esp. in highly technical docs.

Reviewer: This is still not addressed... CA Admin Certificates are never self-signed, so that part of line 109 needs to be removed.

Author: this one is done.

#### TLS CA Root Certificate

> **Description**: Certificate used when making admin requests to the TLS CA.

Reviewer: Not correct. I would copy the text from the Enrollment CA Root Certificate description:

Public Certificate that permits verification of all certificates issued by the TLS CA.

Author: ok

> **Location**: Stored on disk in the TLS CA directory (ca-cert.pem), and copied into the channel configuration to verify identifies for the organization.

Reviewer: ```suggestion
**Location**: Stored on disk in the TLS CA directory (ca-cert.pem), and copied into the channel configuration to verify TLS certificates for the organization.
```

Reviewer: This comment is not addressed yet.

Author: this is done now.

**Impact if expired**: A new TLS CA Root Certificate must be issued. TLS CA Root Certificates are valid for 15 years.

#### TLS CA Admin Certificate

**Description**: Certificate used for admin requests to the TLS CA.

> **Location**: Dependent on implementation:

Reviewer: Colon here is confusing given that there is a Note under it. It would help to put the Note in a box or blockquote.

<pre>
msp
 ├── IssuerPublicKey
 ├── IssuerRevocationPublicKey
 ├── cacerts
 │   └── localhost-7053.pem
 ├── keystore
 │   └── key.pem
 ├── signcerts
 │   └── cert.pem
 └── user
</pre>

> **Impact if expired**: The Fabric Administrator will no longer be able to issue or renew TLS certificates for nodes in the network.

Reviewer: ```suggestion
**Impact if expired**: The Fabric Administrator will no longer be able to register TLS certificates in the TLS CA for nodes in the network.
```

Reviewer: This comment is not addressed yet.

Author: this is done now. was in a "hidden conversation"

> [Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-tls-ca-s-admin)

Reviewer: Every Reference link in this document should indicate what the Reference is to, rather than just say "Reference"

Author: done

### Peer Certificates

> Peer Enrollment Certificates and Peer TLS Certificates are issued for each organization.

Reviewer: ```suggestion
Peer Enrollment Certificates and Peer TLS Certificates are issued for each peer of each organization.
```

Author: A Peer Enrollment Certificate and a Peer TLS Certificate are issued for each peer in an organization.

**Location**: Dependent on implementation:

<pre>
org1ca
└── peer1
    ├── msp
    │    ├── admincerts
    │    │   └── cert.pem
    │    ├── cacerts
    │    │   └── localhost-7053.pem
    │    ├── keystore
    │    │   └── key.pem
    │    ├── signcerts
    │    │   └── cert.pem
    │    └── user
    |── tls
</pre>

**Impact if expired**: Production outage. Peers do not start without a valid Enrollment Certificate.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-peer1)

#### Peer TLS Certificate

**Description**: Authenticates node component communication on the channel.

**Location**: Dependent on implementation:

<pre>
org1ca/
└── peer1
    ├── msp
    └── tls
        ├── cacerts
        ├── keystore
        │   └── key.pem
        ├── signcerts
        │   └── cert.pem
        ├── tlscacerts
        │   └── tls-localhost-7053.pem
        └── user
</pre>

**Impact if expired**: Production outage. No communication to the peer is possible.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-peer1)

### Orderer Certificates

> Orderer Enrollment Certificates and Orderer TLS Certificates are issued for each organization.

Reviewer: ```suggestion
Orderer Enrollment Certificates and Orderer TLS Certificates are issued for each ordering service node of each organization.
```

Reviewer: This comment is not addressed yet.

Author: done.

    ├── keystore
    │   └── key.pem
    ├── signcerts
    │   └── cert.pem
    └── user
</pre>

**Impact if expired**: Transactions can continue to work successfully. Cannot modify channels from a client application or manage the orderer from the console.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-org0-s-admin)

#### Peer Service Organization Channel Admin Certificate

**Description** - Certificate for an organization administrator to manage a peer, including channel and chaincode services.

**Location** - Dependent on implementation:

<pre>
org1ca/
└── org1admin
└── msp
├── admincerts
│     └── cert.pem
├── cacerts
│     └── localhost-7053.pem
├── keystore
│     └── key.pem
├── signcerts
│     └── cert.pem
└── user
</pre>

**Impact if expired**: Transactions can continue to work successfully. Cannot install new smart contracts from a client application or manage the peer from the console.

[Reference](https://hyperledger-fabric-ca.readthedocs.io/en/latest/operations_guide.html#enroll-org1-s-admin)

### Client Certificates

**Description**: Two types of Client Certificates are issued for each organization:

> 1. **Organization Enrollment Certificate** - Authenticates the client identity for communication with peers and orderers.

Reviewer: I would reserve the word "communication" for the TLS Certificate. Maybe say "interactions" here instead.

```suggestion
1. **Organization Enrollment Certificate** - Authenticates the client identity for interactions with peers and orderers.
```

Author: ok.

2. **TLS Certificate** - Authenticates client communications, and only required if mutual TLS is configured.

Client Certificates expire after one year, using the Hyperledger Fabric CA default settings. Client Certificates can be re-enrolled using either command line Hyperledger Fabric CA utilities or the Fabric CA client SDK.

> **Impact if expired**: Client Certificates must be re-enrolled before expiration or the client application will not be able to connect to the Fabric nodes.

Reviewer: ```suggestion
**Impact if expired**: Client Certificates must be re-enrolled before expiration or the client application will not be able to interact with the Fabric nodes.
```

Author: ok. (so if these are different, implies an app can interact without connecting.)

**Enrollment** – An identity based on the registration username and password. The enrollment contains a public/private key pair and an X.509 certificate issued by the organization Certificate Authority (CA). The certificate encodes roles and attributes from the registration.

> **Identity** - A public certificate and its private key used for . The public certificate is the X.509 certificate from the CA, while the private key is issued and stored by the MSP.

Author: to fix.
